build.sh

```diff
@@ -13,14 +13,34 @@ export LC_ALL=en_US.UTF-8
 ARTIFACT_DIRS="${BUILD_ARTIFACTSTAGINGDIRECTORY:-$PWD}"
 mkdir -p "${ARTIFACT_DIRS}/logs"
 
-case $3 in
+has_run_all_tests_trailer() {
+  if [ "${BUILD_REASON:-}" = "PullRequest" ]; then
+    ref="HEAD^2"
+  else
+    ref="HEAD"
+  fi
+  commit=$(git rev-parse $ref)
+  run_all_tests=$(git log -n1 --format="%(trailers:key=run-all-tests,valueonly)" $commit)
+  [[ $run_all_tests == "true" ]]
+}
+
+ALL_TESTS_FILTER="-pr-only"
+LESS_TESTS_FILTER="-main-only"
+
+case ${3:-} in
```

Comment:
```suggestion
case ${3:-main} in
```
would avoid the need for the `|''` later on.

Better yet, you could move this definition up to the top of the file along the lines of:
```
pipeline=${3:-main}
```
I like having a single block near the top of the file that gets all the arguments, so I don't need to read the whole file to find out how many arguments to pass and what they mean.

Reply:
done